hist/histv7/src/RAxis.cxx

```diff
@@ -81,23 +81,24 @@ bool ROOT::Experimental::RAxisEquidistant::HasSameBinBordersAs(const RAxisBase&
 int ROOT::Experimental::RAxisIrregular::GetBinIndexForLowEdge(double x) const noexcept
 {
    // Check in which bin `x` resides
-   const int binIdx = FindBin(x);
+   double fracBinIdx = FindRawBin(x);
+   const int binIdx = std::round(fracBinIdx);
```

Comment:
You shouldn't need to round here, this function will check both edges of the target bin as appropriate.